client/webserver/site/src/css/mixins.scss

```diff
@@ -0,0 +1,92 @@
+@use 'sass:map';
+
+@mixin stylish-overflow {
+  overflow: auto;
+  scrollbar-width: thin;
+  scrollbar-color: #7774 #7771;
+
+  /* Works on Chrome/Edge/Safari */
+  &::-webkit-scrollbar {
+    width: 8px;
+    height: 8px;
+  }
+
+  &::-webkit-scrollbar-track {
+    background-color: var(--body-bg-tertiary);
+  }
+
+  &::-webkit-scrollbar-thumb {
+    background-color: var(--body-bg-tertiary);
+    border-radius: 4px;
+  }
+}
+
+@mixin fill-abs {
+  position: absolute;
+  top: 0;
+  bottom: 0;
+  left: 0;
+  right: 0;
+}
+
+@mixin border($width: 1px, $style: solid, $color: var(--border-color)) {
+  border: $width $style $color;
+}
+
+@mixin border-top($width: 1px, $style: solid, $color: var(--border-color)) {
+  border-top: $width $style $color;
+}
+
+@mixin border-bottom($width: 1px, $style: solid, $color: var(--border-color)) {
+  border-bottom: $width $style $color;
+}
+
+@mixin border-left($width: 1px, $style: solid, $color: var(--border-color)) {
+  border-left: $width $style $color;
+}
+
+@mixin border-right($width: 1px, $style: solid, $color: var(--border-color)) {
+  border-right: $width $style $color;
+}
+
+
+// spacing
+@mixin padding($top: 0, $right: 0, $bottom: 0, $left: 0) {
+  padding: map.get($spacers, $top) map.get($spacers, $right) map.get($spacers, $bottom) map.get($spacers, $left);
+}
+
+@mixin padding-left($padding: 0) {
+  padding-left: map.get($spacers, $padding);
+}
+
+@mixin padding-right($padding: 0) {
+  padding-right: map.get($spacers, $padding);
+}
+
+@mixin padding-bottom($padding: 0) {
+  padding-bottom: map.get($spacers, $padding);
+}
+
+@mixin padding-top($padding: 0) {
+  padding-top: map.get($spacers, $padding);
+}
+
+@mixin margin($top: 0, $right: 0, $bottom: 0, $left: 0) {
+  margin: map.get($spacers, $top) map.get($spacers, $right) map.get($spacers, $bottom) map.get($spacers, $left);
+}
+
+@mixin margin-left($margin: 0) {
+  margin-left: map.get($spacers, $margin);
+}
+
+@mixin margin-right($margin: 0) {
+  margin-right: map.get($spacers, $margin);
+}
+
+@mixin margin-bottom($margin: 0) {
+  margin-bottom: map.get($spacers, $margin);
+}
+
+@mixin margin-top($margin: 0) {
+  margin-top: map.get($spacers, $margin);
```

Comment:
Are we using all of these? Isn't the goal to get rid of this and apply the layout utility classes in markup instead?

Reply:
There are contexts where these are used:
* responsive layouts (e.g. https://github.com/peterzen/dcrdex/blob/normalize-css/client/webserver/site/src/css/wallets.scss#L291) - spacer and border classes don't have breakpoint modifiers
* for elements that aren't individually classed (e.g. `& > form {}`; https://github.com/peterzen/dcrdex/blob/c12b0596adb48746faae789863dee0620fbe743b/client/webserver/site/src/css/forms.scss#L249
* for elements that require other CSS layout rules I used these so that all layout aspects are defined in one place instead of both markup and CSS.

The spacers could actually be gotten rid of with a bit of refactor.

Reply:
Refactored the spacing rules in the markup.  Some of the borders also but kept some of the conditional CSS (in the above contexts) as it's much easier to read/follow.